### PR TITLE
compilation time switch to enable/disable mining improvements

### PR DESCRIPTION
**Pull request by fpelliccioni**

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -117,6 +119,9 @@ def configure(self):
         else:
             self.options["*"].keoken = self.options.keoken
 
+        self.options["*"].mining = self.options.mining
+        self.output.info("Compiling with mining improvements: %s" % (self.options.mining,))
```
Comment (mdallago) [inline conanfile.py]: "improvements" ?? I think it's better to use the word "support" or similar

Comment (dario-ramos) [reply conanfile.py]: If it is a new, separate feature, "support" is good. Otherwise, "enhancements" or "optimizations" sound better

Comment (fpelliccioni) [reply conanfile.py]: They are enhancements...

Comment (fpelliccioni) [reply conanfile.py]: changed!

**Pull request merged**